src/include/atomic.h

```diff
@@ -70,6 +70,17 @@ namespace ceph {
       ceph_spin_unlock(&lock);
       return ret;
     }
+    int cas(T o, T n) {
```

Comment:
should we call it compare_and_swap?

also, is the return value true/value (bool) or the previous value, or the new value?  bool seems least error-prone to me.